src/integration/asset/iothink/IothinkAssetIntegration.ts

```diff
@@ -65,24 +65,67 @@ export default class IothinkAssetIntegration extends AssetIntegration<AssetSetti
 
   private filterConsumptionRequest(asset: Asset, data: any, manualCall: boolean): AbstractCurrentConsumption[] {
     const consumptions: AbstractCurrentConsumption[] = [];
+    // Rename the value property in logs with tag reference
+    for (const dataSet of data.historics) {
+      dataSet.logs.forEach((log) => {
+        log[dataSet.tagReference] = log['value'];
+        delete log['value'];
+      });
+    }
+    // Create helper map to merge the arrays to combined objects
+    const map = new Map();
+    for (const dataSet of data.historics) {
+      dataSet.logs.forEach((item) => map.set(item.timestamp, { ...map.get(item.timestamp), ...item }));
+    }
+    // Create Array with merged consumptions from map
+    const mergedResponse = Array.from(map.values());
+    // Sort the Array according the timestamp
+    mergedResponse.sort((a, b) => a.timestamp - b.timestamp);
     const energyDirection = asset.assetType === AssetType.PRODUCTION ? -1 : 1;
-    if (!Utils.isEmptyArray(data.historics)) {
-      for (let i = 0; i < data.historics[0].logs.length; i++) {
+    if (!Utils.isEmptyArray(mergedResponse)) {
+      for (const mergedConsumption of mergedResponse) {
+        if (typeof mergedConsumption[IothinkProperty.IO_POW_ACTIVE] === 'undefined') {
```

Comment:
IothinkProperty -> IoThinkProperty

Reply:
@LucasBrazi06  
Iothink is consistently written like this in the application. 
I did this because it is the company name and in the end one word.
Should this be changed in the whole Application (Front end and back end?)

Reply:
That's okay, I can live with this ;-).
I close the conversation.